src/ServiceControl/Recoverability/Retrying/RetriesGateway.cs

```diff
@@ -157,21 +157,22 @@ private void StageRetryByUniqueMessageIds(string requestId, RetryType retryType,
                 return;
             }
 
-            var batchDocumentId = retryDocumentManager.CreateBatchDocument(requestId, retryType, messageIds.Length, originator, startTime, last, batchName, classifier);
+            var failedMessageRetryIds = messageIds.Select(FailedMessageRetry.MakeDocumentId).ToArray();
+
+            var batchDocumentId = retryDocumentManager.CreateBatchDocument(requestId, retryType, failedMessageRetryIds, originator, startTime, last, batchName, classifier);
```

Comment:
If SC goes 💥 after this line, the `RetryBatch` will be persisted but not the `FailedMessageRetry` documents.

Then when SC starts up again and adopts this batch, it will move the batch to staging but the `FailedMessageRetry` will still not be created anywhere. Given that I don't see value to these documents, I don't think this is a problem, but perhaps...?

Reply:
:+1: